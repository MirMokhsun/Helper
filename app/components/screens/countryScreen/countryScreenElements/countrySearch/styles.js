import { StyleSheet } from 'react-native';
import Utils from '../../../../../api/utils/utils';

const { width } = Utils.size;

export const styles = StyleSheet.create({
    searchInputContent: {
        flex: 1,
        flexDirection: 'row',
        borderWidth: 1,
        borderColor: "#5A6772",
        borderRadius: 20,
        paddingHorizontal: 10,
        marginHorizontal: 10,
        height: 40,
        width: width - 100,
        backgroundColor: '#E5E5E5',
    },
    svg: {
        width: 24,
        height: 24
    },
    svgContainer: {
        width: 56,
        height: '100%',
        justifyContent: 'center',
        alignItems: 'center',
    },
    textInput: {
        flex: 1,
        fontFamily: 'Roboto-Light',
        fontSize: 14,
        padding: 0,
        marginRight: 10,

    },
    textinputWithTextAlightRight: {
        textAlign: 'right',
        flex: 1,
        fontFamily: 'Roboto-Light',
        fontSize: 14,
        padding: 0,
        marginRight: 10,
    },
});