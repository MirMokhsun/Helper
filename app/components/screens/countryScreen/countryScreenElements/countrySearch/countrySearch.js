import React from 'react';
import { View, TextInput, TouchableOpacity } from 'react-native';
import { styles } from './styles';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { setSearchCountryText } from '../../redux/searchCountryTextAction';
import IconSearchGray from '../../../../commonComponents/svg/Icon_search_gray';
import { localization } from '../../../../../common/localization';

export class CountrySearch extends React.Component {
    constructor(props) {
        super(props);
    }

    UNSAFE_componentWillUnmount() {
        const { setSearchCountryText } = this.props;
        setSearchCountryText("");
    }

    inputChangeText = (text) => {
        const { setSearchCountryText } = this.props;
        setSearchCountryText(text.toLowerCase());
    }

    refFunc = (input) => {
        this.testInput = input
    }

    render() {
        const { language } = this.props;
        return (
            <View style={styles.searchInputContent}>
                <TouchableOpacity onPress={() => { this.testInput.focus() }} testID="BtnIconCountryID" accessibilityLabel="BtnIconCountryID">
                    <View style={styles.svgContainer}>
                        <IconSearchGray />
                    </View>
                </TouchableOpacity>
                <TextInput
                    style={[styles.textInput, language === 'ar' ? styles.textinputWithTextAlightRight : styles.textInput]}
                    underlineColorAndroid='transparent'
                    selectionColor="#5A6772"
                    ref={this.refFunc}
                    onChangeText={this.inputChangeText}
                    placeholderTextColor='#000756'
                    placeholder={localization[language]["Enter your country"]}
                    testID="SearchInputCountryID"
                    accessibilityLabel="SearchInputCountryID"
                />
            </View>
        );
    }
}

const mapStateToProps = state => ({
    language: state.currentLanguage.language,
});
const mapDispatchToProps = dispatch => bindActionCreators({
    setSearchCountryText
}, dispatch);
export default connect(mapStateToProps, mapDispatchToProps)(CountrySearch);
