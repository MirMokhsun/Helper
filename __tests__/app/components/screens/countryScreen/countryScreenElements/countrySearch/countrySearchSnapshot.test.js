import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { CountrySearch } from '../../../../../../../app/components/screens/countryScreen/countryScreenElements/countrySearch/countrySearch';

it('create snapshot CountrySearch', () => {
    const renderer = new ShallowRenderer()
    let toolbarSnapshot = renderer.render(<CountrySearch
        setSearchCountryText={() => { }}
        language ={"ru"}
    />);
    expect(toolbarSnapshot).toMatchSnapshot();
});
