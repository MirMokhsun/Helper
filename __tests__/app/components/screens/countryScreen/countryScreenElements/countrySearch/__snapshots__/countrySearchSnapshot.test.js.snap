// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot CountrySearch 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#E5E5E5",
      "borderColor": "#5A6772",
      "borderRadius": 20,
      "borderWidth": 1,
      "flex": 1,
      "flexDirection": "row",
      "height": 40,
      "marginHorizontal": 10,
      "paddingHorizontal": 10,
      "width": 650,
    }
  }
>
  <TouchableOpacity
    accessibilityLabel="BtnIconCountryID"
    activeOpacity={0.2}
    onPress={[Function]}
    testID="BtnIconCountryID"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "center",
          "width": 56,
        }
      }
    >
      <IconSearchGray />
    </View>
  </TouchableOpacity>
  <TextInput
    accessibilityLabel="SearchInputCountryID"
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Укажите страну"
    placeholderTextColor="#000756"
    rejectResponderTermination={true}
    selectionColor="#5A6772"
    style={
      Array [
        Object {
          "flex": 1,
          "fontFamily": "Roboto-Light",
          "fontSize": 14,
          "marginRight": 10,
          "padding": 0,
        },
        Object {
          "flex": 1,
          "fontFamily": "Roboto-Light",
          "fontSize": 14,
          "marginRight": 10,
          "padding": 0,
        },
      ]
    }
    testID="SearchInputCountryID"
    underlineColorAndroid="transparent"
  />
</View>
`;
